objects/src/accounts/mod.rs

```diff
@@ -71,6 +71,27 @@ impl Account {
         Self { id, vault, storage, code, nonce }
     }
 
+    /// Creates and returns a new account constructed from the specified ID and delta.
+    pub fn from_delta(id: AccountId, delta: &AccountDelta) -> Result<Self, AccountError> {
+        let vault = AssetVault::new(&delta.vault().added_assets)
+            .map_err(|error| AccountError::AssetVaultCreateError { account_id: id, error })?;
+        let storage = AccountStorage::new(
+            delta
+                .storage()
+                .updated_items
+                .iter()
+                .map(|&(index, value)| SlotItem {
+                    index,
+                    slot: StorageSlot::new_value(value),
+                })
+                .collect(),
+        )?;
```

Comment:
This may be fine for now - but we'll also need to handle other storage slot types (e.g., storage map which are implemented in #521). So, I wonder if it would be better to encapsulate this in the `AccountStorageDelta` - e.g., by implementing `TryFrom<AccountStorageDelta> for <AccountStorage>`.